pino.js

```diff
@@ -75,6 +75,28 @@ Object.defineProperty(pinoPrototype, 'levelVal', {
 
 Object.defineProperty(pinoPrototype, '_setLevel', {
   value: function _setLevel (level) {
+    if (typeof level === 'object') {
+      // process and save rules for future offspring
+      // rules is an ordered (more specific first) list of [regexp, level]
+      this.levelRules = Object.keys(level).sort(tools.orderByPrecision).map((pattern) => {
+        return [new RegExp('^' + pattern.replace(/\*/g, '.*?') + '$'), level[pattern]]
+      })
+      this.level = defaultOptions.level
+      // apply rules to existing offspring
+      for (let j = 0, jlen = this.children.length; j < jlen; j++) {
```

Comment:
I think V8 does the `jlen` optimization itself nowadays. And we don't use `let` in Pino; this should be a `var` declaration.